libyul/AsmAnalysis.cpp

```diff
@@ -517,6 +522,26 @@ Scope& AsmAnalyzer::scope(Block const* _block)
 	return *scopePtr;
 }
 
+void AsmAnalyzer::expectValidIdentifier(YulString _identifier, SourceLocation const& _location)
+{
+	// NOTE: the leading dot case is handled by the parser not allowing it.
```

Comment:
I'm starting to feel like the parser should allow it and it should be a nice error here.

Reply:
Maybe... depends on the amount of hassle that'd be... at some point we need some major cleanup for the yul parsing process anyways, e.g. we still reuse the same token list and scanner for solidity and yul and need weird workarounds because of it, etc.

Reply:
Why shouldn't the parser to all this?

Reply:
I think all of it should be done at the same place eventually. I prefer the analyser because it can provide much nicer error messages.